[FEATURE] Afficher les CGUs suivant la langue de l'utilisateur dans Pix Orga (PIX-2354).

yahyayouness opened the pull request:
## :unicorn: Problème
La page des conditions générales d'utilisation n'est pas traduite dans Pix Orga.


## :robot: Solution

Compte tenu de la quantité de traductions (12 pages) et de la problématique de l'édition de cette page et de son habillage. On a opté pour la création de deux composants, un pour la version FR et un autre pour la version EN. L'avantage est de garder la flexibilité en terme d'édition et de ne pas mettre de balises html (h2 h3 p li bold ... ) dans le JSON. L'inconvénient est qu'il y aurait autant de composant par version. Ceci dit, concernant la version arabe, cette solution semble la mieux adaptée.
Autre solution non testée, est de rajouter un sous dossier translations et d'avoir un fichier dédié.

Comme il existe une EPIC versioning CGUs avec une migration vers Prismic. Et que la version du site en espagnol n'arrive que dans deux ans, on a jugé que cette solution est la mieux adaptée pour le contexte actuel.

Une correction de l'affichage en responsive en mobile a été effectuée.
Un changement au niveau des titres pour améliorer l'accessibilité.

## :100: Pour tester

### JDD
Supprimer l'accpetation des CGU
`update "users" set "pixOrgaTermsOfServiceAccepted" = false;`

### Scénario
Se connecter à PixOrga
- RA
   - [français](https://orga-pr2950.review.pix.fr)
   - [anglais](https://orga-pr2950.review.pix.org/cgu?lang=en)
- local
   - [français](http://localhost.fr:4201/connexion)
   - [anglais](http://localhost.org:4201/connexion)

Vérifier le contenu des CGUs affichées.
Les valider, vérifier qu'elles ne sont plus réaffichées lors de la reconnexion.

yahyayouness committed:
Add translation for accept button in terms of service page

pix-service commented:
I'm deploying this PR to these urls:

- App: https://app-pr2950.review.pix.fr
- Orga: https://orga-pr2950.review.pix.fr
- Certif: https://certif-pr2950.review.pix.fr
- Admin: https://admin-pr2950.review.pix.fr
- API: https://api-pr2950.review.pix.fr/api/

 Please check it out!

yahyayouness committed:
Add component term of service with FR locale

octo-topi commented on orga/app/components/terms-of-service-en.hbs:
```diff
@@ -0,0 +1,707 @@
+    <h3>Article 1. Preamble</h3>
+    <p>
+      The Pix “Groupement d’Intérêt Public” (French public interest group; GIP) approved by the decree of 27 April 2017
+      giving approval of the agreement establishing the public interest group ‘Pix’, with registered office at 156
+      Boulevard
+      Haussmann, 75008 Paris (France), developed the Pix platform within the framework of a public project in
```
**question**
On garde Boulevard Hausmann alors que les mentions [légales](https://pix.fr/mentions-legales/) utilisent Ardennes ?

```suggestion
      giving approval of the agreement establishing the public interest group ‘Pix’, with registered office at 
      21 Rue des Ardennes, 75019 Paris (France), developed the Pix platform within the framework of a public project in
```

octo-topi commented on orga/app/components/terms-of-service-en.hbs:
```diff
@@ -0,0 +1,704 @@
+      and collect and analyse the results from the Pix platform.
+    </p>
+    <p>
+      The Pix Orga platform is accessible at the following URLs: : orga.pix.fr and orga.pix.org.
```
**suggestion**
```suggestion
      The Pix Orga platform is accessible at the following URLs: orga.pix.fr and orga.pix.org.
```

octo-topi commented on orga/app/components/terms-of-service-fr.hbs:
```diff
@@ -0,0 +1,720 @@
+      </p>
+      <p>
+        <b>Etape 2.</b> L’utilisateur se connecte à son compte utilisateur aux moyens de ses identifiants de connexion,
+        étant rappelé quee ces derniers sont personnels et confidentiels.
```
**suggestion**
```suggestion
        étant rappelé que ces derniers sont personnels et confidentiels.
```

octo-topi commented on orga/app/components/terms-of-service-en.hbs:
```diff
@@ -0,0 +1,704 @@
+      unsolicited advertising or spam or chain emails.
+    </p>
+
+    <h3>Article 10. Sécurité</h3>
```
**suggestion**
```suggestion
    <h3>Article 10. Security</h3>
```

octo-topi reviewed (approved):
L'IHM est conforme au document fourni dans JIRA
J'ai fait une dizaaine [de remarques dans JIRA](https://1024pix.atlassian.net/browse/PIX-2354?focusedCommentId=19152) concernant le document et l'existant. 

Je suis pour merger tel quel, et faire un deuxième ticket s'il y a des changements dans les entrants.

yahyayouness committed:
Add component term of service with EN locale

yahyayouness committed:
Split term of service into two components for en/fr version

yahyayouness commented:
> L'IHM est conforme au document fourni dans JIRA
> J'ai fait une dizaaine [de remarques dans JIRA](https://1024pix.atlassian.net/browse/PIX-2354?focusedCommentId=19152) concernant le document et l'existant.
> 
> Je suis pour merger tel quel, et faire un deuxième ticket s'il y a des changements dans les entrants.

Thanks pour tes retours, c'est pris en compte

AndreiaPena commented on orga/app/components/terms-of-service-en.hbs:
```diff
@@ -0,0 +1,704 @@
+  <hr class="terms-of-service-form__line">
+
+  <div class="terms-of-service-form__text">
+    <h3>Article 1. Preamble</h3>
```
**Accessibilité** : l’ordre des titres n'est pas respecté, on commence ici directement par un h3. Il faut un h1, un h2 puis viendra le h3. A modifier en anglais et en français.

<img width="269" alt="Capture d’écran 2021-05-07 à 10 40 07" src="https://user-images.githubusercontent.com/58915422/117423091-9cd5be80-af20-11eb-8105-3d9f867997f9.png">

J'ai testé sur Firefox avec l'outil Wave.

MelanieMEB replied:
Votre `h1` est de préférence unique, donc ça serait : Condition Générale d'utilisation de Pix-Orga (à mettre en sr-only si vous ne le voulez pas).
Le premier "Article" sera donc en h2.

yahyayouness replied:
On a utilisé les titres h1,h2,h3 d'une manière hiérarchique with @AndreiaPena , n'hésitez pas à refaire une passe.

AndreiaPena commented:
**Suggestions CSS pour le responsive** : 
Voilà le rendu de cette page en mobile actuellement 👇

<img width="594" alt="Capture d’écran 2021-05-07 à 11 46 41" src="https://user-images.githubusercontent.com/58915422/117431937-13c38500-af2a-11eb-9653-2d7ca127d4c2.png">

Je propose des retouches CSS sur `terms-of-service.scss` avec des `@include device-is('mobile')`:

- Ajouter un width au bloc blanc pour qu'il soit entièrement visible.
```
.terms-of-service-form {
  @include device-is('mobile') {
    width: 95%;
  }
}
```

Mettre le margin de la zone de texte à 0
```
&__text {
    @include device-is('mobile') {
      margin: 0;
    }
  }
```

Je propose qu'en mobile, les boutons "Annuler" et "J'accepte" soit l'un sous l'autre, sinon le bouton "J'accepte" n'affiche pas tout son wording.
```
&__actions {
    @include device-is('mobile') {
      flex-direction: column;
      align-items: center;
      margin: 15px 10px 0 10px;
    }
  }
```

Pour le bouton "J'accepte" on ajoute la ligne bleu en extérieur pour le focus et hover, pour plus de visibilité
```
&__submit {
    @include device-is('mobile') {
      margin: 10px 0 0 0;
    }

    &:focus,
    &:hover,
    &:focus-within {
      border: 2px solid white;
      box-shadow: 0 0 0 2px $blue;
      padding: 0;
    }
  }
```
Ce qui donnera ceci 👇
<img width="521" alt="Capture d’écran 2021-05-07 à 11 57 54" src="https://user-images.githubusercontent.com/58915422/117433163-79644100-af2b-11eb-98f5-4228a262a464.png">


S'il y a aussi moyen de réduire la taille de l'image pix Orga aussi 🙂

yahyayouness committed:
[Accessibility] Use hiearchical heading for improve screen reader

A heading describes the content that follows it, much like a news headline. When arriving at a new page, sighted users gravitate toward headings to quickly find what they want on the page. Screen reader and other assistive technology users can also skip from heading to heading.

yahyayouness committed:
Adapt terms of service page to be responsive in mobile device

yahyayouness commented:
> **Suggestions CSS pour le responsive** :
> Voilà le rendu de cette page en mobile actuellement 👇
> 
> <img alt="Capture d’écran 2021-05-07 à 11 46 41" width="594" src="https://user-images.githubusercontent.com/58915422/117431937-13c38500-af2a-11eb-9653-2d7ca127d4c2.png">
> 
> Je propose des retouches CSS sur `terms-of-service.scss` avec des `@include device-is('mobile')`:
> 
> * Ajouter un width au bloc blanc pour qu'il soit entièrement visible.
> 
> ```
> .terms-of-service-form {
>   @include device-is('mobile') {
>     width: 95%;
>   }
> }
> ```
> 
> Mettre le margin de la zone de texte à 0
> 
> ```
> &__text {
>     @include device-is('mobile') {
>       margin: 0;
>     }
>   }
> ```
> 
> Je propose qu'en mobile, les boutons "Annuler" et "J'accepte" soit l'un sous l'autre, sinon le bouton "J'accepte" n'affiche pas tout son wording.
> 
> ```
> &__actions {
>     @include device-is('mobile') {
>       flex-direction: column;
>       align-items: center;
>       margin: 15px 10px 0 10px;
>     }
>   }
> ```
> 
> Pour le bouton "J'accepte" on ajoute la ligne bleu en extérieur pour le focus et hover, pour plus de visibilité
> 
> ```
> &__submit {
>     @include device-is('mobile') {
>       margin: 10px 0 0 0;
>     }
> 
>     &:focus,
>     &:hover,
>     &:focus-within {
>       border: 2px solid white;
>       box-shadow: 0 0 0 2px $blue;
>       padding: 0;
>     }
>   }
> ```
> 
> Ce qui donnera ceci 👇
> <img alt="Capture d’écran 2021-05-07 à 11 57 54" width="521" src="https://user-images.githubusercontent.com/58915422/117433163-79644100-af2b-11eb-98f5-4228a262a464.png">
> 
> S'il y a aussi moyen de réduire la taille de l'image pix Orga aussi 🙂

Thanks pour le partage de  l'outil https://responsively.app/ , tes suggestions réparent l'affichage mobile :)

Anne-Gaelle-S commented on orga/app/styles/pages/authenticated/terms-of-service.scss:
```diff
@@ -33,7 +37,7 @@
     color: $grey-100;
     font-family: $roboto;
-    font-size: 18px;
+    font-size: 1.15em;
```
**question**
C'est plutôt en `rem` l'unité ici. (em est pas très recommandé pour la taille de la police)

Anne-Gaelle-S commented on orga/app/styles/pages/authenticated/terms-of-service.scss:
```diff
@@ -52,17 +56,50 @@
+.terms-of-service-form-text {
+
+  &__article-title {
+    font-size: 1.15em;
   }
 
+  &__article-subtitle {
+    font-size: 1em;
```
**suggestion**
Idem ici, c'est voulu les `em` ? En général on ne les utilises pas pour la police ... 
Mieux vaut priviligié les `rem` :)
